src/adb.js

```diff
@@ -487,6 +488,220 @@ class Adb {
         );
     });
   }
+
+  // AW : return the file size of a complete folder
+  getFileSize(file) {
+    var _this = this;
+    return new Promise(function(resolve, reject) {
+      _this.getAdbState().then(stdout => {
+        if (stdout == 1) {
+          // Recovery
+          _this
+            .shell("du -shk " + file)
+            .then(stdout => {
+              stdout = parseFloat(stdout);
+              _this.log("FileSize is " + stdout + " Ko");
+              resolve(stdout);
+            })
+            .catch(reject);
+        } else if (stdout == 2) {
+          // device (type : used, size)
+          _this
+            .shell("df -hBK " + file + " --output=used|tail -n1")
+            .then(stdout => {
+              stdout = parseFloat(stdout);
+              _this.log("FileSize is " + stdout + " Ko");
+              resolve(stdout);
+            })
+            .catch(reject);
+        }
+      });
+    }).catch(e => {
+      _this.log(e);
+      reject;
+    });
+  }
+
+  // AW : Return the status of the device (0 : bootloader, 1 : recovery, 2 : device ...)
+  getAdbState() {
+    var _this = this;
+    return new Promise(function(resolve, reject) {
+      _this
+        .execCommand(["get-state"])
+        .then(stdout => {
+          _this.log("Device state is " + stdout);
+          switch (stdout) {
+            case "device":
+              resolve(2);
+            case "bootloader":
+              resolve(0);
+            case "recovery":
+              resolve(1);
+            default:
+              resolve(3); // Unknown
+          }
+        })
+        .catch(e => {
+          // Try with recovery command
+          _this.log("Unknown error :" + e);
+          reject;
+        });
+    });
+  }
+
+  // AW : Backup file "srcfile" from the phone to "destfile" localy
+  // TO-DO : This function should be moved to devices.js
```

Comment:
I'd be ok with having this in here, but it should be renamed to something like `createUbuntuBackup()`.